app/src/main/java/com/kleshchin/danil/filemanager/ListAdapter.java

```diff
@@ -0,0 +1,105 @@
+package com.kleshchin.danil.filemanager;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+
+/**
+ * Created by Danil Kleshchin on 11.05.2017.
+ */
+
+public class ListAdapter extends BaseAdapter {
+
+    @NonNull
+    private ArrayList<File> file_;
+    @NonNull
+    private Context context_;
+
+    public ListAdapter(Context context, File file) {
```

Comment:
Nullness annotations can be used not only with class fields declaration. Android studio should warn you about not annotated constructor parameters in this case. Read more about nullness annotations, they are very useful in java